### ISSUE TITLE
Improve integration with Notifications add-on

### ISSUE THREAD

**Issue by artemy63** [jmix-framework/jmix#596]

**Comment by Flaurite** [comment]
## Summary
Re-implements (redesigned) the BPM integration with Notifications for Jmix 2: a running BPM process can now send in-app / email notifications through the Notifications add-on, configured entirely in the process modeler. The feature is opt-in and a no-op when the add-on is absent, so existing processes are unaffected.

## What was done
Two ways to attach notifications to a process, both authored visually in the modeler (bpm-flowui), no code required:

1) "Send Notification" service task — a standalone step (mirrors the existing Send Email task), shown on the canvas with a bell icon.
2) "Notifications" panel on a user task — notifications bound to the task lifecycle events create, assignment, complete, delete. (Custom/signal-driven cases are modeled as a boundary event → a Send Notification task.)

Each notification supports:

- Recipients — assignee, candidateUsers, candidateGroups (fanned out to members via `UserGroupService`, incl. USERS/ROLES groups), initiator, or an expression.
- Content — inline subject/body with Flowable ${...} expressions, or the body rendered from a jmix-messagetemplates template (content type plain/html; a template's own type wins).
- Channels — multi-value in-app / email (email offered only when the email add-on is present).
- Notification type — optional; drives the in-app icon/category.

The notification definitions are stored inline in the BPMN XML as repeated `<jmix:notification>` extension elements, extracted at runtime, resolved to recipients, rendered, and dispatched to the add-on's `NotificationManager`. Dispatch is fire-and-forget — a failing expression or delivery error never aborts the process nor suppresses the other notifications.

## How to use
Prerequisites: add the jmix-notifications add-on (plus jmix-messagetemplates for template bodies, and an email add-on/configured Emailer for the email channel).

In the modeler:

1. On a task, open the replace menu → Send Notification Task (or select a user task and use its Notifications section).
1. In the Notifications grid, Create a notification and fill: recipient type, channels, content type, subject, body; optionally pick a message template or a notification type.
1. Save the diagram, deploy, and run the process — notifications are delivered as the process reaches the task / fires the event.

Example of the stored definition on a user task:
```
<userTask id="approve" name="Approve request">
  <extensionElements>
    <jmix:notification event="assignment" recipientType="assignee" channels="in-app,email" contentType="plain">
      <jmix:subject>Task assigned: ${task.name}</jmix:subject>
      <jmix:body>Hello ${context.recipientUser.username}, you have a new task.</jmix:body>
    </jmix:notification>
  </extensionElements>
</userTask>
```

Expressions available in subject/body include process variables (`${orderNumber}`), object navigation and method calls (`${order.date.getYear()}`), and process info via the context (`${context.processDefinitionName}`, `${context.taskName}`, `${context.recipientUser.username}`).

## Compatibility
Opt-in and backward-compatible: no configuration means no notifications, and with the add-ons absent the feature is an inert no-op — existing BPM applications are unaffected.

## Test plan
1. In the modeler add notifications on a Send Notification task and a user task → save → reopen and confirm round-trip;
1. Deploy and run a process → confirm in-app/email delivery with `${...}` substitution;
1. Try different types of recipients. Make sure that notification is sent to every type.
1. Start a BPM app without jmix-notifications/jmix-messagetemplates and confirm it starts cleanly and processes run (notifications skipped).

**Comment by SergeiAksenov2** [comment]
1. Should the Notification be saved without a selected Channel?
2. Message Template pop-up form should be larger and be resizeble.

**Comment by Flaurite** [comment]
@SergeiAksenov2

1) I've made channel field required. By default, without selected channel the mechanism sets in-app channel. The disabling sending some notifications should be separated issue here that will created by user request.
2) Created issue: https://github.com/jmix-framework/jmix/issues/5567

**Comment by SergeiAksenov2** [comment]
Tested on:
Jmix version: 3.1.999-SNAPSHOT
Jmix Studio Plugin Version: 3.1.SNAPSHOT8646-261
IntelliJ version: IntelliJ IDEA 2026.1.4
Successful.
